### PR TITLE
[Enhance] Add ignore_index to CrossEntropyLoss

### PR DESCRIPTION
**Pull request by AronLin**
The `CrossEntropyLoss` has no interface `ignore_index`, which makes it inconvenient to complete the segmentation task.

### CODE REVIEW

**Comment by AronLin** [comment]
I will check the training and inference processes of HTC.

**Comment by AronLin** [comment]
kindly ping @jshilong.

**Comment by AronLin** [comment]
The training and inference processes of Mask R-CNN are ok.

File: mmdet/models/losses/cross_entropy_loss.py [hunk 1]
```diff
@@ -23,12 +24,20 @@ def cross_entropy(pred,
         avg_factor (int, optional): Average factor that is used to average
             the loss. Defaults to None.
         class_weight (list[float], optional): The weight for each class.
+        ignore_index (int | None): The label index to be ignored.
+            If None, it will be set to default value. Default: -100.
```
Comment (jshilong) [inline mmdet/models/losses/cross_entropy_loss.py:28]: I am confused about this default value

File: mmdet/models/losses/cross_entropy_loss.py [hunk 2]
```diff
@@ -23,12 +24,20 @@ def cross_entropy(pred,
         avg_factor (int, optional): Average factor that is used to average
             the loss. Defaults to None.
         class_weight (list[float], optional): The weight for each class.
+        ignore_index (int | None): The label index to be ignored.
```
Comment (jshilong) [inline mmdet/models/losses/cross_entropy_loss.py:27]: `ignore_index` may be confusing, no keyword in the name to indicate it  is only related  to `label`

Comment (jshilong) [reply mmdet/models/losses/cross_entropy_loss.py:27]: Is it right the type `int? Does it only support ignore one class?

File: mmdet/models/losses/cross_entropy_loss.py [hunk 3]
```diff
@@ -60,7 +74,8 @@ def binary_cross_entropy(pred,
                          weight=None,
                          reduction='mean',
                          avg_factor=None,
-                         class_weight=None):
+                         class_weight=None,
+                         ignore_index=255):
```
Comment (jshilong) [inline mmdet/models/losses/cross_entropy_loss.py:78]: 255 is so confusing

File: mmdet/models/losses/cross_entropy_loss.py (hunk 2 above)
Comment (AronLin) [reply mmdet/models/losses/cross_entropy_loss.py:27]: ignore_index is a key of F.cross_entropy_loss(). It only supports ignore one class. Usually, this number does not belong to the classes of a model.

File: mmdet/models/losses/cross_entropy_loss.py (hunk 1 above)
Comment (AronLin) [reply mmdet/models/losses/cross_entropy_loss.py:28]: -100 is the default value of ignore_index in F.cross_entropy_loss.

File: mmdet/models/losses/cross_entropy_loss.py (hunk 3 above)
Comment (AronLin) [reply mmdet/models/losses/cross_entropy_loss.py:78]: In this function, we do not use ignore_index in the F.cross_entropy_loss.
So I refer to the setting of mmsegmentation, setting it to 255 by default.

**Review by ZwwWayne — APPROVED** [review]
